pygeos/_geometry.pyx

```diff
@@ -81,6 +199,14 @@ def get_parts(object[:] array):
     return parts, index
 
 
+cdef _deallocate_arr(void* handle, np.intp_t[:] arr, Py_ssize_t last_geom_i):
+    """Deallocate a temporary geometry array to prevent memory leaks"""
+    cdef Py_ssize_t i = 0
+
+    for i in range(last_geom_i):
+        GEOSGeom_destroy_r(handle, <GEOSGeometry *>arr[i])
```

Comment:
I know this was just centralized and not otherwise new, but given recent other changes to handle freeing things we shouldn't so as to avoid segfaults, I wonder if we should first check that `arr[i]` is != `NULL` before calling `GEOSGeom_destroy_r` on it - just to be sure?